capstone/src/main/java/com/google/sps/servlets/YoutubeServlet.java

```diff
@@ -38,27 +39,24 @@
   private static final String KEY = "API_KEY";
   private static final String NUM_RESULTS = "5";
   private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+  private String postRequestBodyData;
+  private String channelId;
+  private String videoId; 
   OkHttpClient client = new OkHttpClient();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String ChannelId = request.getParameter("channelId");
-    String videoId = request.getParameter("videoId");
-    String completeUrl = (ChannelId != null) ? 
-      (BASE_URL + "&allThreadsRelatedToChannelId=" + ChannelId + "&maxResults=" + NUM_RESULTS + "&key=" + KEY) : 
-        (BASE_URL + "&videoId=" + videoId + "&maxResults=" + NUM_RESULTS + "&key=" + KEY);
-    String output = get(completeUrl);
+    String completeUrl = (postRequestBodyData.length() == 24) ? 
```

Comment:
Rather than checking if the string is 24 characters to determine if it's a channel id or video id, could you instead have the POST request take a string in JSON format the specifies either the channelId or videoId?  This might mean changing your JavaScript code to do either:
```
const response = await fetch(URL, {
    method: 'POST',
    headers: {'Content-Type': 'application/json',},
    body: JSON.stringify({'channelId': channelId})
});
```
or
```
const response = await fetch(URL, {
    method: 'POST',
    headers: {'Content-Type': 'application/json',},
    body: JSON.stringify({'videoId': videoId}) // Note this is using a different name in the JSON object
});
```

I think this would be more clear to the reader of the code, and would be handle edge cases where a videoId might be 24 characters

Reply:
I've modified the YoutubeServlet and it's server calls to support a JSON input in "update youtube servlet with json support". The new input contains the type of id along with the id itself.